File: CardGameDBSites.Frontend/app/services/requirements/PointsRequirement.ts
import type { CardDetailApiModel, CardsQueryFilterClauseApiModel } from "~/api/default";
import type { IRequirement } from "./IRequirement";
import RequirementType from "./RequirementType";
import { GetCardValue } from "~/helpers/CardHelper";

export default class PointsRequirement implements IRequirement {
  RequirementType: RequirementType = RequirementType.Points;

  IsValid(cards: CardDetailApiModel[], config: Record<string, any>): boolean {
    const ability = config.ability as string;
    const min = config.min as number | undefined;
    const max = config.max as number | undefined;

    // Calculate total points from all cards
    let totalPoints = 0;
    for (const card of cards) {
      const pointValue = GetCardValue<string | number>(card, ability);
      if (pointValue === null) {
        // If a card doesn't have the points attribute, treat it as 0
        continue;
      }
      const parsedPoints = typeof pointValue === "number" ? pointValue : Number(pointValue);
      if (!Number.isFinite(parsedPoints)) {
        continue;
      }
      totalPoints += parsedPoints;
    }

    // Check if total is within the range
    if (min !== undefined && totalPoints < min) {
      return false;
    }
    if (max !== undefined && totalPoints > max) {
      return false;
    }

    return true;
  }

  ToFilters(
    cards: CardDetailApiModel[],
    config: Record<string, any>,
  ): CardsQueryFilterClauseApiModel[] | undefined {
    // Points requirements can't be easily converted to filters without knowing
    // the currently selected cards, so we return undefined
    return undefined;
  }
}

File: SkytearHorde.Entities/Requirements/PointsRequirement.cs
using SkytearHorde.Entities.Constants;
using SkytearHorde.Entities.Models.Business;
using SkytearHorde.Entities.Interfaces;

namespace SkytearHorde.Entities.Requirements
{
    public class PointsRequirement : ISquadRequirement
    {
        public string Alias => RequirementConstants.PointsAlias;

        private readonly string _ability;
        private readonly decimal? _min;
        private readonly decimal? _max;

        public PointsRequirement(string ability, decimal? min, decimal? max)
        {
            _ability = ability;
            _min = min;
            _max = max;
        }

        public bool IsValid(Card[] cards)
        {
            decimal totalPoints = 0;

            foreach (var card in cards)
            {
                var pointValue = card.GetMultipleCardAttributeValue(_ability);
                if (pointValue is null || pointValue.Length == 0)
                {
                    // If a card doesn't have the points attribute, treat it as 0
                    continue;
                }

                if (decimal.TryParse(pointValue[0], out var parsedPoints))
                {
                    totalPoints += parsedPoints;
                }
            }

            if (_min != null && totalPoints < _min)
                return false;
            if (_max != null && totalPoints > _max)
                return false;
            return true;
        }
    }
}
